python/triqs_cthyb/solver.py

```diff
@@ -145,6 +147,23 @@ def solve(self, **params_kw):
         # (only supported for G_tau, to permit compatibility with dft_tools)
         if perform_post_proc and (self.last_solve_parameters["measure_G_tau"] == True):
 
+            if self.last_solve_parameters["measure_density_matrix"]:
+                # we have the density matrix, so we will compute the analytic moments
+                # and orbital occupations
+                self.orbital_occupations = orbital_occupations(self.density_matrix,
+                                                               self.gf_struct,
+                                                               self.h_loc_diagonalization
+                                                               )
+
+                self.Sigma_moments = high_frequency_moments(self.density_matrix,
+                                                 self.h_loc_diagonalization,
+                                                 self.gf_struct,
+                                                 self.last_solve_parameters['h_int']
+                                                 )
+
+                self.Sigma_Hartree = self.Sigma_moments[0]
+                
+
             # Fourier transform G_tau to obtain G_iw
             for bl, g in self.G_tau:
                 bl_size = g.target_shape[0]
```

Comment:
Do we also want to use the moments to improve the Fourier trafo? We could set `known_moments[2,...]` right?